core/src/hir/defs.rs

```diff
@@ -19,6 +20,18 @@ pub enum TypeDef<'tcx> {
     OutStruct(&'tcx OutStructDef),
     Opaque(&'tcx OpaqueDef),
     Enum(&'tcx EnumDef),
+    Trait(&'tcx TraitDef),
```

Comment:
issue: I don't think it's meaningful for traits to live under TypeDef. They should be a separate thing entirely, resolved via separate paths.